python/tests/operations/abs.py

```diff
@@ -1,210 +1,77 @@
+            )
 
-        block_2 = TensorBlock(
-            values=np.array([[-1, -2], [-3, 4], [5, -6]]),
-            samples=Labels(["samples"], np.array([[0], [2], [7]])),
-            components=[],
-            properties=Labels.arange("properties", 2),
-        )
-        block_2.add_gradient(
-            "parameter",
-            data=np.array(
-                [
-                    [[-10, 11], [-12, 13]],
-                    [[-14, -15], [10, -11]],
-                    [[-12, -13], [14, -15]],
-                ]
-            ),
-            samples=Labels(
-                ["sample", "positions"],
-                np.array([[0, 1], [1, 1], [2, 1]]),
-            ),
-            components=[
-                Labels.arange("components", 2),
-            ],
-        )
+        blocks.append(new_block)
 
-        block_res1 = TensorBlock(
-            values=np.array([[1, 2], [3, 5]]),
-            samples=Labels(["samples"], np.array([[0], [2]])),
-            components=[],
-            properties=Labels.arange("properties", 2),
-        )
-        block_res1.add_gradient(
-            "parameter",
-            data=np.array([[[-6, 1], [7, 2]], [[8, -3], [9, 4]]]),
-            samples=Labels(["sample", "positions"], np.array([[0, 1], [1, 1]])),
-            components=[
-                Labels.arange("components", 2),
-            ],
-        )
-        block_res2 = TensorBlock(
-            values=np.array([[1, 2], [3, 4], [5, 6]]),
-            samples=Labels(["samples"], np.array([[0], [2], [7]])),
-            components=[],
-            properties=Labels.arange("properties", 2),
-        )
-        block_res2.add_gradient(
-            "parameter",
-            data=np.array(
-                [
-                    [[10, -11], [12, -13]],
-                    [[14, -15], [-10, -11]],
-                    [[-12, 13], [14, 15]],
-                ]
-            ),
-            samples=Labels(
-                ["sample", "positions"],
-                np.array([[0, 1], [1, 1], [2, 1]]),
-            ),
-            components=[
-                Labels.arange("components", 2),
-            ],
+    return TensorMap(keys=tensor.keys, blocks=blocks)
+
+
+@pytest.fixture
+def tensor_map_result(tensor_map_complex):
+    blocks = []
+    for _, block in tensor_map_complex:
+        new_block = TensorBlock(
+            values=np.abs(block.values),
+            samples=block.samples,
+            components=block.components,
+            properties=block.properties,
         )
-        keys = Labels(names=["key_1", "key_2"], values=np.array([[0, 0], [1, 0]]))
-        A = TensorMap(keys, [block_1, block_2])
-        tensor_abs = equistore.abs(A)
-        tensor_result = TensorMap(keys, [block_res1, block_res2])
 
-        assert equistore.equal(tensor_result, tensor_abs)
+        for parameter, gradient in block.gradients():
+            new_block.add_gradient(
+                parameter=parameter,
+                data=-gradient.data,
```

Comment:
Right now since all the values of block.values are negative having -gradient.data works fine, but maybe we can do another test for the general case.

Reply:
Okay, I'll do that.

Reply:
Thanks @DivyaSuman14! We can also take a look together and write the test.